www/js/survey/multilabel/MultiLabelButtonGroup.tsx

```diff
@@ -0,0 +1,132 @@
+  const $ionicPopup = getAngularService("$ionicPopup");
+  const $ionicPopover = getAngularService("$ionicPopover");
+
+  useEffect(() => {
+    console.log("During initialization, trip is ", trip);
+    ConfirmHelper.inputParamsPromise.then((ip) => setInputParams(ip));
+  }, []);
+
+  // to mark 'inferred' labels as 'confirmed'; turn yellow labels blue
+  function verifyTrip() {
+    for (const inputType of ConfirmHelper.INPUTS) {
+      const inferred = trip.finalInference[inputType];
+      // TODO: figure out what to do with "other". For now, do not verify.
+      if (inferred?.value && !trip.userInput[inputType] && inferred.value != "other")
+        store(inputType, inferred.value, false);
+    }
+  }
+
+  function openPopover(e, inputType) {
+    let popoverPath = 'templates/diary/'+inputType.toLowerCase()+'-popover.html';
+    const scope = createScopeWithVars({inputParams, choose});
+    $ionicPopover.fromTemplateUrl(popoverPath, {scope}).then((pop) => {
+      closePopover = () => pop.hide();
+      pop.show(e);
+    });
+  }
+
+  function choose(inputType, chosenLabel, wasOther=false) {
+    if (chosenLabel && chosenLabel != "other") {
+      store(inputType, chosenLabel, wasOther);
+    } else {
+      const scope = createScopeWithVars({other: {text: ''}});
+      $ionicPopup.show({
+        scope,
+        title: t("trip-confirm.services-please-fill-in", { text: inputType.toLowerCase() }),
+        template: '<input type="text" ng-model="other.text">',
+        buttons: [
+          {
+            text: t('trip-confirm.services-cancel')
+          },
+          {
+            text: `<b>${t('trip-confirm.services-save')}</b>`,
+            type: 'button-positive',
+            onTap: () => choose(inputType, scope.other.text, true)
+          }
+        ]
+      });
+    }
+  };
+
+  function store(inputType, chosenLabel, isOther) {
+    if (isOther) {
+      /* Let's make the value for user entered inputs look consistent with our other values
+       (i.e. lowercase, and with underscores instead of spaces) */
+      chosenLabel = chosenLabel.toLowerCase().replace(" ", "_");
+    }
+    const inputDataToStore = {
+      "start_ts": trip.start_ts,
+      "end_ts": trip.end_ts,
+      "label": chosenLabel,
+    };
+
+    const storageKey = ConfirmHelper.inputDetails[inputType].key;
+    window['cordova'].plugins.BEMUserCache.putMessage(storageKey, inputDataToStore).then(() => {
+      closePopover?.();
+      repopulateTimelineEntry(trip._id.$oid);
+      console.debug("Successfully stored input data "+JSON.stringify(inputDataToStore));
+    });
+  }
+
+  const inputKeys = Object.keys(trip.inputDetails);
+  return (
+    <View style={{flexDirection: 'row', alignItems: 'center'}}>
+      <View style={{width: 'calc(100% - 20px)'}}>
+        {inputKeys.map((key, i) => {
+          const input = trip.inputDetails[key];
+          const inputIsConfirmed = trip.userInput[input.name];
```

Comment:
do we still have the distinction between `trip.userInput` and `trip.user_input`? I have frequently felt that we should rename one or the other just to avoid confusion among future maintainers.

Reply:
I didn't rename them. I've also thought it would be a good idea; for additions as well.

Currently, the server gives us `user_input` and `additions`, and the phone organizes these into `userInput` and `additionsList`

Maybe the phone can do `userInputList` and `additionsList` -- or a better descriptive name if we can think of one

Reply:
*TODO later*

Reply:
`aggUserInput` and `aggAdditions` ?
I'm thinking it's basically an aggregation of processed input(s) from server and unprocessed input(s) on phone